### PR TITLE
DOMA-3445 Refactor billing domain -- Add register billing receipts funciton

### PR DESCRIPTION
**Pull request by toplenboren**
> Hi, this PR is still WIP. Comments would be welcome, but will be removed and review would be re-requested once PR is ready

# Problem

1. Billing integration send too many requests to the CRM
2. Billing integrations may be developed by other software engineering teams, rendering the domain uncontrollable
3. Right now engineering team that is developing an integration with billing should implement diff finding and data enrichment functions. It costs time and money for other teams! 

# Solution

Create a unified mutation which will handle validation, parsing and diff creation functions.

# Documentation

Full documentation on this feature, covering all corner cases would be available once the feature is ready for production

# TODO:
- [x] Basic mutation logic
- [x] Access Rights
- [x] Handle simple create cases 
- [x] Context validation 
- [x] Limit
- [ ] Refactoring!
- [x] Category validation
- [x] Tests with complex and update cases
- [x] Address validation
- [ ] Bulk implementation in generator functions
- [ ] Documentation on corner cases
- [ ] Billing Property domain cleanup
- [ ] Billing Account domain cleanup

### CODE REVIEW

File: apps/condo/domains/billing/schema/RegisterBillingReceiptsService.js [hunk 1]
```diff
@@ -0,0 +1,295 @@
+/**
+ * Generated by `createservice billing.RegisterBillingReceiptsService --type mutations`
+ */
+
+const _ = require('lodash')
```
[the remark this answers is not in the document]
Comment (toplenboren) [reply apps/condo/domains/billing/schema/RegisterBillingReceiptsService.js:5]: find collides with keystoneUtils find :-) thats why I used `_`

File: apps/condo/domains/billing/utils/serverSchema/index.js [hunk 2]
```diff
@@ -55,6 +56,20 @@ async function exportPayments (context, data) {
 }
 
 const BillingCategory = generateServerUtils(BillingCategoryGQL)
+async function registerBillingReceipts (context, data) {
+    if (!context) throw new Error('no context')
+    if (!data) throw new Error('no data')
+    if (!data.sender) throw new Error('no data.sender')
+    // TODO(codegen): write registerBillingReceipts serverSchema guards
```
Comment (pahaz) [inline apps/condo/domains/billing/utils/serverSchema/index.js:63]: !

File: apps/condo/domains/billing/utils/testSchema/index.js [hunk 3]
```diff
@@ -664,6 +667,46 @@ async function makeClientWithIntegrationAccess () {
...
+    const sender = { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+
+    const attrs = {
+        dv: 1,
+        sender,
+        ...args,
+    }
+    const { data, errors } = await client.mutate(REGISTER_BILLING_RECEIPTS_MUTATION, { data: attrs })
+
+    throwIfError(data, errors)
```
Comment (pahaz) [inline apps/condo/domains/billing/utils/testSchema/index.js:682]: it's better to add third argument

File: apps/condo/domains/billing/utils/testSchema/index.js [hunk 4]
```diff
@@ -664,6 +667,46 @@ async function makeClientWithIntegrationAccess () {
     return client
 }
 
+async function registerBillingReceiptsByTestClient(client, args) {
```
Comment (pahaz) [inline apps/condo/domains/billing/utils/testSchema/index.js:670]:
```suggestion
async function registerBillingReceiptsByTestClient(client, extraAttrs) {
```

File: apps/condo/domains/billing/utils/testSchema/index.js [hunk 5]
```diff
@@ -664,6 +667,46 @@ async function makeClientWithIntegrationAccess () {
...
+    if (!client) throw new Error('no client')
+    if (!args) throw new Error('no data')
+    const sender = { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+
+    const attrs = {
+        dv: 1,
+        sender,
+        ...args,
```
Comment (pahaz) [inline apps/condo/domains/billing/utils/testSchema/index.js:678]: do you want to use createRegisterBillingReceiptsPayload here?

**Comment by toplenboren** [comment]
Opened up a new PR from a branch with normal name
